### PR TITLE
Add city card entry and related screen

### PR DESCRIPTION
**Pull request by ching56**

### CODE REVIEW

**Comment by ching56** [comment]
- [ ] Replace placeholder image with photos and  QRcode

File: src/components/ClaimCardItem.tsx [hunk 1]
```diff
@@ -56,18 +63,22 @@ class ClaimCardItem extends React.Component<IProps, IStates> {
   }
 
   render() {
+    const hasCover =
+      this.props.title === 'Taipei Card' || this.props.title === 'Tainan Card';
+    const title = hasCover ? '' : this.props.title;
+    const source = hasCover
+      ? coverSources[this.props.title!]
+      : {
```
Comment (chihsuan) [inline src/components/ClaimCardItem.tsx]: We should consider to move this into `coverSources` as it should be.

File: src/navigators/AppNavigator.tsx [hunk 2]
```diff
@@ -159,6 +163,39 @@ const ClaimsStack = createStackNavigator(
         title: 'Asset Detail',
       }),
     },
+    ClaimDetailCityCard: {
+      screen: ClaimDetailCityCardScreen,
+      navigationOptions: () => ({
+        headerStyle: {},
```
Comment (chihsuan) [inline src/navigators/AppNavigator.tsx]: Could we use a variable to store this default object?

File: src/navigators/AppNavigator.tsx [hunk 3]
```diff
@@ -159,6 +163,39 @@ const ClaimsStack = createStackNavigator(
...
+        title: 'About Owner',
+        headerStyle: {},
+        headerTintColor: '',
+        headerTitleStyle: '',
+      }),
+    },
+    ClaimCityCardCountry: {
+      screen: ClaimCityCardCountryScreen,
+      navigationOptions: () => ({
+        headerStyle: {},
```
Comment (chihsuan) [inline src/navigators/AppNavigator.tsx]: Could we use a variable to store this default object?

File: src/navigators/AppNavigator.tsx [hunk 4]
```diff
@@ -159,6 +163,39 @@ const ClaimsStack = createStackNavigator(
...
+        headerTintColor: '',
+        headerTitleStyle: '',
+      }),
+    },
+    ClaimCityCardCountryEvents: {
+      screen: ClaimCityCardCountryEventsScreen,
+      navigationOptions: () => ({
+        headerStyle: {},
```
Comment (chihsuan) [inline src/navigators/AppNavigator.tsx]: Could we use a variable to store this default object?

File: src/screens/ClaimCityCardCountryEventsScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+  state = {
+    filterText: '',
+    openSearchBar: false,
+  };
+
+  searchBarOffset: number = 80;
+
+  onScroll = (event: NativeSyntheticEvent<NativeScrollEvent>) => {
+    const { nativeEvent } = event;
+    if (nativeEvent.contentOffset.y < -30 && !this.state.openSearchBar) {
```
Comment (chihsuan) [inline src/screens/ClaimCityCardCountryEventsScreen.tsx]: Please try to make this more readable.

File: src/screens/ClaimCityCardCountryEventsScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+          return { openSearchBar: true };
+        }
+        return null;
+      });
+    }
+
+    if (
+      this.state.openSearchBar &&
+      !this.state.filterText &&
+      nativeEvent.contentOffset.y > this.searchBarOffset + 10
```
Comment (chihsuan) [inline src/screens/ClaimCityCardCountryEventsScreen.tsx]: Please try to make this more readable.

File: src/screens/ClaimCityCardCountryEventsScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+      </View>
+    );
+  }
+
+  render() {
+    const area: BiiMe.types.TCityCardArea = this.props.navigation.getParam(
+      'area',
+      { coverUri: '', name: 'Wrong!', events: [] },
+    );
+    const filteredEvents = area.events.filter(event =>
```
Comment (chihsuan) [inline src/screens/ClaimCityCardCountryEventsScreen.tsx]: Please use memorize approach to calculate filtered results.

File: src/screens/ClaimCityCardCountryEventsScreen.tsx [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+            <TextInput
+              placeholder="Search"
+              value={this.state.filterText}
+              style={myStyles.searchBarTextInput}
+              onChangeText={filterText => this.setState({ filterText })}
+            />
+            <TouchableOpacity onPress={() => this.setState({ filterText: '' })}>
+              <MaterialIcon
+                name="cancel"
+                size={this.state.filterText ? 20 : -1}
```
Comment (chihsuan) [inline src/screens/ClaimCityCardCountryEventsScreen.tsx]: If you want to hide this, please just not render this or use style to hide it.

File: src/screens/ClaimCityCardOwnerScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+      <Text style={myStyles.title}>{title}</Text>
+      <Text style={myStyles.value}>{value}</Text>
+    </View>
+  );
+  render() {
+    const { name = '', nationalId = '', license = '', birth = '' } = this.props
+      .navigation.state.params!.data as BiiMe.types.TCityCardOwner;
+    return (
+      <View style={myStyles.container}>
+        {this.renderRow('Name:', name)}
```
Comment (chihsuan) [inline src/screens/ClaimCityCardOwnerScreen.tsx]: Could we use a component for the row instead of function.

File: src/store/index.ts [hunk 10]
```diff
@@ -1,26 +1,41 @@
...
 
-const initialState = {};
 const persistedReducer = persistReducer(persistConfig, reducers);
 const sagaMiddleware = createSagaMiddleware();
 
 export default () => {
   const store = createStore(
     persistedReducer,
-    initialState,
+    {},
```
Comment (chihsuan) [inline src/store/index.ts]: Why do we change this?

**Comment by ching56** [comment]
I have resolved some comments, 
And others @TseHang will deal with them with another PR later,
Merge for now to move on, thanks!

File: src/store/index.ts (hunk 10 above)
Comment (TseHang) [reply src/store/index.ts]: Oh it's a typo. However, initialState is originally a `{}`

File: src/screens/ClaimCityCardCountryEventsScreen.tsx (hunk 7 above)
Comment (TseHang) [reply src/screens/ClaimCityCardCountryEventsScreen.tsx]: Do you mean like this?

```
const filteredEvents = area.events.filter(this.handleFilterText);
```

**Pull request merged**